### PR TITLE
Product Directory Refresh

### PR DESCRIPTION
**Pull request by hturan**
### Summary
This PR gives a visual and functional refresh to the Products page (https://developers.cloudflare.com/products), including iconography, product descriptions, and the ability to search.

### Screenshots (optional)
Screenshot 2024-11-12 at 15 43 53
Screenshot 2024-11-12 at 15 44 02
Screenshot 2024-11-12 at 15 44 16
Screenshot 2024-11-12 at 15 45 29
Screenshot 2024-11-12 at 18 31 27

### Documentation checklist
- [x] The [documentation style guide](https://developers.cloudflare.com/style-guide/) has been adhered to.

### CODE REVIEW

File: src/components/ProductCatalog.tsx
Comment (hturan) [inline src/components/ProductCatalog.tsx:1]: Lots of room for us to de-dupe this and `ModelCatalog` in the future. Left it be for now until we tackle a third instance of this pattern.

File: src/pages/products.astro [hunk 1]
```diff
@@ -1,181 +1,49 @@
 ---
-import StarlightPage from "@astrojs/starlight/components/StarlightPage.astro";
 import { getCollection, type CollectionEntry } from "astro:content";
-import { ThreeCardGrid, LinkCard } from "~/components";
+// @ts-ignore virtual module
+import iconCollection from "virtual:astro-icon";
```
Comment (hturan) [inline src/pages/products.astro:4]: I got 90% of the way through this work and then realised we have no good way to render icon / SVGs in `react`, and `astro-icon` does some interesting integration/optmimization under the hood. This code here replicates a little of what the `Icon` component does (https://github.com/natemoo-re/astro-icon/blob/531db17ec5367e22df7958d8f7abdeb880d8342b/packages/core/components/Icon.astro#L92) but just passes optimised SVG path data (as a string) through to React to render.

Comment (hturan) [reply src/pages/products.astro:4]: I originally had this in the `jsx` component, but due to how Astro works it was trying to import `node` modules into the client-side and breaking.

Comment (hturan) [reply src/pages/products.astro:4]: I also tried directly importing the icons (using `import` URLs), but you can't style SVG when used as an `img` `src`.

**Review by KimJ15 — APPROVED** [review]

File: src/pages/products.astro [hunk 2]
```diff
@@ -1,181 +1,49 @@
...
-	}
-
-	getGroupFilters()?.forEach((input) => {
-		input.addEventListener("change", filterProducts);
-	});
-
-	function initProducts() {
-		const params = new URLSearchParams(window.location.search);
-
-		const groups = params.get("product-group");
```
Comment (KianNH) [inline src/pages/products.astro:164]: It looks like we're missing this functionality now - we rely on it on the homepage using links like `/products/?product-group=Developer+platform`

File: src/components/index.ts [hunk 3]
```diff
@@ -48,7 +48,6 @@ export { default as ResourcesBySelector } from "./ResourcesBySelector.astro";
 export { default as RuleID } from "./RuleID.astro";
 export { default as SpotlightAuthorDetails } from "./SpotlightAuthorDetails.astro";
 export { default as Stream } from "./Stream.astro";
-export { default as ThreeCardGrid } from "./ThreeCardGrid.astro";
```
Comment (KianNH) [inline src/components/index.ts:51]: Can we delete the component file itself too? I think the products grid was the only usage of it.

File: src/pages/products.astro [hunk 4]
```diff
@@ -1,181 +1,49 @@
...
-									href={product.data.product.url}
-								/>
-							</div>
-						);
-					})
-				}
-			</ThreeCardGrid>
-		</div>
-	</div>
+	<ProductCatalog products={products} client:load />
```
Comment (KianNH) [inline src/pages/products.astro]: CI is failing due to a type error on `products` here: https://github.com/cloudflare/cloudflare-docs/actions/runs/11820899830/job/32934322983?pr=18133#step:6:46

**Review by KianNH — CHANGES_REQUESTED** [review]
Awesome work, the styling is perfect! This looks great.

I found a few small issues which are hopefully quick fixes.

File: src/pages/products.astro (hunk 2 above)
Comment (hturan) [reply src/pages/products.astro:164]: Ah, sorry! Totally missed this, my bad. I'll add it via stetting initial state in React.

File: src/pages/products.astro (hunk 4 above)
Comment (hturan) [reply src/pages/products.astro]: Thanks! I've made a more specific `ProductData` type for the `ProductCatalog` (including icons and groups).

**Review by KianNH — APPROVED** [review]
❤️

**Pull request merged**